data_override/include/data_override_impl.inc

```diff
@@ -1,188 +1,112 @@
-logical                                         :: reproduce_null_char_bug = .false. !> Flag indicating
-                                                   !! to reproduce the mpp_io bug where lat/lon_bnd were
-                                                   !! not read correctly if null characters are present in
-                                                   !! the netcdf file
-
-namelist /data_override_nml/ debug_data_override, grid_center_bug, reproduce_null_char_bug
-
+  type(data_type)                            :: default_table
+  type(override_type), dimension(max_array)  :: override_array !< to store processed fields
+  type(override_type)                        :: default_array
+  logical                                    :: debug_data_override
+  logical                                    :: grid_center_bug = .false.
+  logical                                    :: reproduce_null_char_bug = .false. !> Flag indicating
+                                                !! to reproduce the mpp_io bug where lat/lon_bnd were
+                                                !! not read correctly if null characters are present in
+                                                !! the netcdf file
 
-public :: data_override_init, data_override, data_override_unset_domains
-public :: data_override_UG
+  namelist /data_override_nml/ debug_data_override, grid_center_bug, reproduce_null_char_bug
 
-contains
-function count_ne_1(in_1, in_2, in_3)
-  logical, intent(in)  :: in_1, in_2, in_3
-  logical :: count_ne_1
+  public :: DATA_OVERRIDE_INIT_IMPL_, DATA_OVERRIDE_UNSET_ATM_, DATA_OVERRIDE_UNSET_OCN_, &
+          & DATA_OVERRIDE_UNSET_LND_, DATA_OVERRIDE_UNSET_ICE_, DATA_OVERRIDE_0D_IMPL_, &
+          & DATA_OVERRIDE_2D_IMPL_, DATA_OVERRIDE_3D_IMPL_, DATA_OVERRIDE_UG_1D_IMPL_, &
+          & DATA_OVERRIDE_UG_2D_IMPL_
 
-  count_ne_1 = .not.(in_1.NEQV.in_2.NEQV.in_3) .OR. (in_1.AND.in_2.AND.in_3)
-end function count_ne_1
+  contains
 
-!> @brief Assign default values for default_table, get domain of component models,
-!! get global grids of component models.
-!! Users should call data_override_init before calling data_override
-!!
-!! This subroutine should be called in coupler_init after
-!! (ocean/atmos/land/ice)_model_init have been called.
-!!
-!! data_override_init can be called more than once, in one call the user can pass
-!! up to 4 domains of component models, at least one domain must be present in
-!! any call
-!!
-!! Data_table is initialized here with default values. Users should provide "real" values
-!! that will override the default values. Real values can be given using data_table, each
-!! line of data_table contains one data_entry. Items of data_entry are comma separated.
-subroutine data_override_init(Atm_domain_in, Ocean_domain_in, Ice_domain_in, Land_domain_in, Land_domainUG_in)
+subroutine DATA_OVERRIDE_INIT_IMPL_(Atm_domain_in, Ocean_domain_in, Ice_domain_in, Land_domain_in, Land_domainUG_in)
```

Comment:
This routine needs documentation

Reply:
Doxygen comment added in 2d6bfd6da5004ab00e7be60e8e85f13cbfa96827.